tests/api/test_statements_post.py

```diff
@@ -0,0 +1,229 @@
+"""
+Tests for the POST statements endpoint of the Ralph API.
+"""
+import base64
+import json
+import os
+from uuid import uuid4
+
+import bcrypt
+import pytest
+from elasticsearch import Elasticsearch
+from elasticsearch.client import IndicesClient
+from fastapi.testclient import TestClient
+
+from ralph.api import app
+from ralph.defaults import APP_DIR
+
+ES_TEST_HOSTS = os.environ.get("RALPH_ES_TEST_HOSTS", "http://localhost:9200").split(
+    ","
+)
+ES_TEST_INDEX = "statements"
+ES_CLIENT = Elasticsearch(ES_TEST_HOSTS)
+ES_INDICES_CLIENT = IndicesClient(ES_CLIENT)
+
+client = TestClient(app)
+
+
+# pylint: disable=invalid-name
+def setup_auth(fs):
+    """
+    Set up the credentials file and return the credentials that need to be passed
+    through headers to authenticate the request.
+    """
+    credential_bytes = base64.b64encode("ralph:admin".encode("utf-8"))
+    credentials = str(credential_bytes, "utf-8")
+
+    auth_file_path = APP_DIR / "auth.json"
+    fs.create_file(
+        auth_file_path,
+        contents=json.dumps(
+            [
+                {
+                    "username": "ralph",
+                    "hash": bcrypt.hashpw(b"admin", bcrypt.gensalt()).decode("UTF-8"),
+                    "scopes": ["ralph_test_scope"],
+                }
+            ]
+        ),
+    )
+
+    return credentials
```

Comment:
As it is used in the GET statements endpoint tests, we can make a fixture instead of duplicating this function?